feat(vfs): 添加路径遍历和目录切换的权限检查

xboHodx opened the pull request:
- 新增permission模块处理inode权限检查
- 在路径遍历和chdir/fchdir系统调用中添加权限验证
- 实现do_fchmodat文件权限修改功能

<!-- CURSOR_SUMMARY -->
---

> [!NOTE]
> Add UNIX-style permission checks to path traversal and chdir/fchdir, implement chmod via fchmodat, and only fallback `arch_prctl` to common handler on EINVAL; update tests.
> 
> - **VFS / Permissions**:
>   - Add `vfs/permission.rs` with `PermissionMask` and `Cred::{inode_permission,check_chdir_permission}` including capability overrides.
>   - Enforce execute/search permission during path traversal in `IndexNode::do_lookup_follow_symlink`.
> - **Syscalls**:
>   - `chdir`/`fchdir`: validate directory and permissions via `check_chdir_permission` before switching CWD.
>   - Implement `do_fchmodat`: update only permission bits, preserve file type, and apply via `set_metadata`.
> - **Arch**:
>   - `arch_prctl`: call x86_64 handler first; fallback to common handler only on `EINVAL`.
> - **Tests**:
>   - Enable `chdir_test`, `fchdir_test`; blocklist `FchdirTest.PermissionDenied`.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 8faea7fcaf1abb870ac76e592c44f96eaa3d8914. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>
<!-- /CURSOR_SUMMARY -->

xboHodx committed:
feat(vfs): 添加路径遍历和目录切换的权限检查

- 新增permission模块处理inode权限检查
- 在路径遍历和chdir/fchdir系统调用中添加权限验证

xboHodx committed:
feat(filesystem): 实现do_fchmodat文件权限修改功能

xboHodx committed:
feat(permission)

xboHodx committed:
fmt

xboHodx committed:
fix(arch_prctl): 优化arch_prctl错误处理逻辑

xboHodx committed:
fmt

fslongjin commented on kernel/src/filesystem/vfs/permission.rs:
```diff
@@ -0,0 +1,164 @@
+    }
+
+    // 所有检查都失败
+    Err(SystemError::EACCES)
```
> 类似的逻辑指的是？是在内核中有相同作用的模块了吗？还是说这些位运算的逻辑？

指的是位运算。这些mask之类的，
![Uploading image.png…]()
这个结构体里面就有

xboHodx committed:
Merge branch 'master' of github.com:xboHodx/DragonOS into fix/syscall/chdir

xboHodx committed:
fix(fs/permission): 重构权限位计算逻辑，启用测例

- 重构权限位计算逻辑使用ModeType常量
- 更新gvisor测试白名单添加chdir相关测试

xboHodx committed:
Merge branch 'master' of github.com:xboHodx/DragonOS into fix/syscall/chdir

xboHodx committed:
暂时block掉gvisor的FchdirTest.PermissionDenied测试用例

fslongjin committed:
Merge remote-tracking branch 'origin/master' into yangshengjie-fix/syscall/chdir

fslongjin committed:
refactor(vfs): 将权限检查函数重构为Cred结构体方法

- 将inode_permission及相关辅助函数从独立函数重构为Cred结构体方法
- 更新sys_chdir和sys_fchdir调用权限检查的新接口
- 优化代码组织结构，减少模块间依赖

Signed-off-by: longjin <longjin@DragonOS.org>
